pylint/config.py

```diff
@@ -594,11 +594,69 @@ def parse(self, to_parse, config):
                     type_ = definition.get('type')
                     validator = VALIDATORS.get(type_, lambda x: x)
                     value = validator(value)
-                config.set_option(option, value)
+                configuration.set_option(option, value)
```

Comment:
I need to reuse most of the parsing code in IniFileParser.write, so moved it to another method.